worlds/yoshisisland/Regions.py

```diff
@@ -0,0 +1,222 @@
+from typing import List, Dict, Tuple
+from BaseClasses import MultiWorld, Region, Entrance, Location
+from .Locations import LocationData
+from .level_logic import YoshiLogic
+from .setup_bosses import BossReqs
+
+class YILocation(Location):
+    game: str = "Yoshi's Island"
+    LevelID: int
+
+def __init__(self, player: int, name: str = " ", address: int = None, parent=None, LevelID: int = None):
+    super().__init__(player, name, address, parent)
+    self.LevelID = LevelID
+
+
+def create_regions(multiworld: MultiWorld, player: int, locations: Tuple[LocationData, ...], location_cache: List[Location], world, boss_order: list, level_location_list: list, luigi_pieces: int):

```

Comment:
```suggestion
if TYPE_CHECKING:
    from . import YIWorld


def create_regions(world: "YIWorld", locations: Tuple[LocationData, ...]):
    multiworld = world.multiworld
    player = world.player
    location_cache = world.location_cache
    boss_order = world.boss_order
    luigi_pieces = world.luigi_pieces
```
If you're passing in world already, there is no need to pass in anything that you can just get from world.
This should also be adjusted in YoshiLogic, connectStartingRegion, BossReqs, and anything else that has world and something you can get off of world.